src/clean-core/allocator.cc

```diff
@@ -0,0 +1,199 @@
+#include <clean-core/allocator.hh>
+
+#include <cstdlib>
+#include <cstring>
+
+#include <clean-core/assert.hh>
+#include <clean-core/utility.hh>
+
+namespace
+{
+struct stack_alloc_header
+{
+    size_t padding;
+};
+
+template <class T>
+[[nodiscard]] T align_up_masked(T value, size_t mask)
+{
+    return (T)(((size_t)value + mask) & ~mask);
+}
+
+template <class T>
+[[nodiscard]] T align_up(T value, size_t alignment)
+{
+    return align_up_masked(value, alignment - 1);
+}
+
+inline std::byte* align_up_with_header(std::byte* head, size_t align, size_t header_size)
```

Comment:
inline in a .cc?

Reply:
removed (these were from a header originally)